dash/src/util/BenchmarkParams.cc

```diff
@@ -189,7 +189,9 @@ void BenchmarkParams::print_pinning()
       << std::setw( 5)     << "core"
       << std::setw( 5)     << "CPU"
       << '\n';
-  for (global_unit_t unit{0}; unit < dash::size(); ++unit) {
+  for (global_unit_t unit{0};
+       unit < static_cast<dash::global_unit_t>(dash::size());
```

Comment:
This cast does not look right.

Reply:
The problem here is actually that `dash::size()` returns a `size_t` while `dash::team_unit_t` and `dash::global_unit_t` are 16-bit signed integers. So to fix that we should return an unsigned 16-bit integer from `dash::size`. However, this is a separate pull request.

Reply:
Actually, the unit IDs in DASH are 32-bit (we currently only have 24bit in the `dart_gptr_t` iirc because of the flags field that is not used). But I agree, we should consider fixing this discrepancy for 0.4.0.